src/bundle/Resources/public/js/scripts/fieldType/ezmedia.js

```diff
@@ -0,0 +1,136 @@
+            const nameContainer = preview.querySelector('.ez-field-edit-preview__file-name');
+            const sizeContainer = preview.querySelector('.ez-field-edit-preview__file-size');
+            const files = [].slice.call(event.target.files);
+            const fileSize = this.formatFileSize(files[0].size);
+
+            this.fieldContainer.querySelector('.ez-field-edit__data').setAttribute('hidden', true);
+            preview.removeAttribute('hidden');
+
+            nameContainer.innerHTML = files[0].name;
+            nameContainer.title = files[0].name;
+            sizeContainer.innerHTML = fileSize;
+            sizeContainer.title = fileSize;
+
+            // it breaks the rendering flow and prevents from blocking
+            // rendering a whole fieldtype container
+            // while loading video source
+            window.setTimeout(this.updateMediaSource.bind(this, files[0]), 100);
+
+            btnRemove.addEventListener('click', this.handleRemoveFile, false);
+        }
+
+        updateMediaSource(file) {
+            const preview = this.fieldContainer.querySelector(SELECTOR_PREVIEW);
+            const videoUrl = URL.createObjectURL(file);
+
+            preview.querySelector('.ez-field-edit-preview__action--preview').href = videoUrl;
+            preview.querySelector(SELECTOR_MEDIA).setAttribute('src', videoUrl)
+        }
+
+        showMediaPreview() {
+            const mediaWrapper = this.fieldContainer.querySelector(SELECTOR_MEDIA_WRAPPER);
+
+            mediaWrapper.classList.remove(CLASS_MEDIA_WRAPPER_LOADING);
+        }
+
+        showMediaLoadingScreen() {
+            const mediaWrapper = this.fieldContainer.querySelector(SELECTOR_MEDIA_WRAPPER);
+
+            mediaWrapper.classList.add(CLASS_MEDIA_WRAPPER_LOADING);
+        }
+
+        handleRemoveFile(event) {
+            super.handleRemoveFile(event);
+
+            this.showMediaLoadingScreen();
+        }
+
+        init() {
+            super.init();
+
+            const preview = this.fieldContainer.querySelector(SELECTOR_PREVIEW);
+            const video = preview.querySelector(SELECTOR_MEDIA);
+
+            video.addEventListener('canplay', this.showMediaPreview.bind(this), false);
+        }
+    }
+
+    [...document.querySelectorAll(SELECTOR_FIELD)].forEach(fieldContainer => {
+        const validator = new EzMediaValidator({
+            classInvalid: 'is-invalid',
+            fieldSelector: fieldContainer,
+            eventsMap: [
+                {
+                    selector: '[type="file"]',
+                    eventName: 'change',
+                    callback: 'validateInput',
+                    invalidStateSelectors: [SELECTOR_FIELD],
+                    errorNodeSelectors: ['.ez-field-edit__label-wrapper'],
+                }, {
+                    selector: '.ez-field-edit-preview__info',
+                    eventName: 'click',
```

Comment:
You can listen to event `change` on checkbox and you would not need to set checked in the callback, just add class to the label

Reply:
The `change` event didn't fire at all, that's why I decided to handle `click` events.